vector/v.cluster/testsuite/cluster_test.py

```diff
@@ -0,0 +1,242 @@
+            type="vector",
+            name="test_points,clustered",
+            flags="f",
+        )
+
+    def setUp(self):
+        self.temp_files = []
+        self.temp_points = self.create_temp_file(
+            "1 1\n1 2\n2 1\n2 2\n3 3\n11 11\n11 12\n12 11\n12 12\n50 50\n50 51\n51 50\n51 51\n100 100"
+        )
+
+        self.temp_3d = self.create_temp_file(
+            "1 1 5\n1 2 7\n2 1 8\n2 2 0\n11 11 10\n11 12 10\n12 11 10\n12 12 10\n50 50 20\n50 51 20\n51 50 20\n51 51 20\n100 100 30"
+        )
+
+        self.runModule(
+            "v.in.ascii",
+            input=self.temp_points,
+            format="point",
+            separator="space",
+            output="test_points",
+            overwrite=True,
+        )
+
+        self.runModule(
+            "v.in.ascii",
+            input=self.temp_3d,
+            format="point",
+            z=3,
+            flags="z",
+            separator="space",
+            output="test_points_3d",
+            overwrite=True,
+        )
+
+    def tearDown(self):
+        """Removes all temporary files created during the tests."""
+        for temp_file in self.temp_files:
+            os.remove(temp_file)
+
+    def create_temp_file(self, content):
+        """Creates a temporary file with the given content and returns its path."""
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as temp_file:
+            temp_file.write(content)
+            temp_file_name = temp_file.name
+        self.temp_files.append(temp_file_name)
+        return temp_file_name
+
+    def get_cluster_info(self, map_name):
+        # Export the clustered points to ASCII format
+        ascii_output = grass.read_command(
+            "v.out.ascii", layer=2, input=map_name, format="point", separator="comma"
+        )
+
+        # print(ascii_output)
+        # Parse the ASCII output to extract cluster IDs
+        clusters = {}
+
+        for line in ascii_output.splitlines():
+            if line.strip():  # Skip empty lines
+                parts = line.split(",")
+                if len(parts) >= 3:
+                    x, y, cluster_id = float(parts[0]), float(parts[1]), int(parts[2])
+                    if cluster_id != 0:  # Skip noise points
+                        if cluster_id not in clusters:
+                            clusters[cluster_id] = []
+                        clusters[cluster_id].append((x, y))
+        return clusters
+
+    def get_noise_points(self, map_name):
+        ascii_output = grass.read_command(
+            "v.out.ascii", layer=2, input=map_name, format="point", separator="comma"
+        )
+
+        noise_points = []
+        for line in ascii_output.splitlines():
+            if line.strip():
+                parts = line.split(",")
+                if len(parts) >= 3:
+                    cluster_id = int(parts[2])
+                    if cluster_id == 0:
+                        noise_points.append(cluster_id)
+
+        return noise_points
+
```

Comment:
Can you merge this into one function, it seems somewhat inefficient to run v.out.ascii twice.